### DISCUSSION TITLE
[General] Can we consider all accounts in zkSync as smart contract accounts?

### DISCUSSION THREAD

**Discussion by PatrickAlphaC** [zkSync-Community-Hub/zksync-developers#492] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

I have recently come across the [DefaultAccount](https://github.com/Cyfrin/zksync-contracts/blob/main/zksync-contracts/l2/system-contracts/DefaultAccount.sol) in the zkSync contracts codebase. 

The zkSync docs say that zkSync has two kinds of accounts, EoAs and smart contracts, but if all EoAs are actually this implementation of smart contract wouldn't we consider all EoA _also_ smart contract accounts? 

### What is the specific issue or error you're encountering?

see above

### Can you share the error messages or logs you're receiving, if any?

see above

### Have you made any recent changes to the contract before encountering this issue?

see above

### Are there any external libraries or contracts that your contract interacts with?

see above

### Can you provide the relevant portions of your contract code where the issue is occurring?

see above

### Have you tried to isolate the problem, and if so, what were the results?

see above

### What steps have you already taken to try to resolve the issue?

see above

**Comment by dutterbutter** [comment, marked as answer]
@PatrickAlphaC that is correct, you can consider all accounts on zkSync to be smart accounts.
